overlord/overlord.go

```diff
@@ -78,6 +78,7 @@ func New() (*Overlord, error) {
 	}
 	o.snapMgr = snapMgr
 	o.stateEng.AddManager(o.snapMgr)
+	snapMgr.Init(s)
```

Comment:
we need to decide what to do about Init? anyway calling Init is a stateEng task not of the overlord